### PR TITLE
HZN-1178: Describe installation of NetFlow v5 protocol

### PR DESCRIPTION
**Pull request by indigo423**
Describe how to install NetFlow v5 feature using Karaf Console and how to configure the Listener. Explain what configuration is required to persist date with the NetFlow v5 adapter to ElasticSearch. Add some troubleshooting hints for operation.

* JIRA: http://issues.opennms.org/browse/HZN-1178

### CODE REVIEW

**Comment by indigo423** [comment]
@RangerRick I have cherry-picked NMS-9735 from release-21.0.1 which is required for this PR. Is this the right thing to do?
@j-white Happy for your feedback about the NetFlow v5 troubleshooting section :D

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+[[telemetryd-netflow5-troubleshooting]]
+===== NetFlow v5 Troubleshooting
+
+.Verify in Karaf Console if the `flows` feature is installed and started
+[source]
+----
+list | grep -i fail
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc]: If the feature is not installed, this will show nothing - the same as if it were installed successfully.

I think using `feature:list | grep flows` would be a better approach here.

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+----
+
+The `Geocoder Nomatim` should be listed as `Failure` which is ok.
+
+.Test connectivity to ElasticSearch with curl with a valid response (use -u <elasticUser>) if authentication is enabled
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc]: I'd rephrase this as:
Test the connectivity to ElasticSearch using curl

And add a tip bellow with:
use -u <elasticUser> if authentication is enabled

**Comment by indigo423** [comment]
thank you for review, went through your comments and updated working branch to just have this commit in the PR.

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-adapter.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+| Class Name          | `org.opennms.netmgt.telemetry.adapters.netflow.Netflow5Adapter`
+|===
+
+====== Parameters
+
+.Adapter specific parameters for the Netflow5Adapter
+[options="header, autowidth"]
+|===
+| Parameter        | Description                                                       | Required | Default value
+| `script`         | Full path to the script used to handle the JTI messages           | required | (none)
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-adapter.adoc]: The Netflow adapter does not use a script. There are no parameters as far as I know.

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
+
+[[telemetryd-netflow5-protocol]]
+==== NetFlow Version 5
+
+link:https://www.cisco.com/c/en/us/td/docs/net_mgmt/netflow_collection_engine/3-6/user/guide/format.html[NetFlow] is an application that provides the ability to collect IP network traffic from network devices and administrators can determine and analyze the network communication.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc]: `is an application` should probably be `is a protocol`

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+IMPORTANT: The NetFlow v5 requires a running ElasticSearch 5.6+ to persist the NetFlow data.
+
+By default the persistence is not installed by default and can be enabled through the Karaf Console with:
+
+[source]
+----
+$ ssh -p 8101 admin@localhost
+...
+admin@opennms()> feature:install flows
+admin@opennms()> config:property-set name Netflow-5
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc]: These additional properties are not required for the `flows` feature. They are ignored in OpenNMS, and only used on Minion.

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+[[telemetryd-netflow5-troubleshooting]]
+===== NetFlow v5 Troubleshooting
+
+.Verify in Karaf Console if the `flows` feature is installed and started
+[source]
+----
+feature:list | grep flows
+----
+
+The `Geocoder Nomatim` should be listed as `Failure` which is ok.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-troubleshooting.adoc]: This is irrelevant now that we grep for `flows` vs `fail`.

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-adapter.adoc [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+|===
+| Class Name | `org.opennms.netmgt.telemetry.adapters.netflow.Netflow5Adapter`
+|===
+
+====== Parameters
+
+.Adapter specific parameters for the Netflow5Adapter
+[options="header, autowidth"]
+|===
+| Parameter | Description | Required | Default value
```
Comment (indigo423) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-adapter.adoc]: @mvrueden is not done and he does not tell me what configuration parameters we have here :D

File: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+IMPORTANT: The NetFlow v5 requires a running ElasticSearch 5.6+ to persist the NetFlow data.
+
+By default the persistence is not installed by default and can be enabled through the Karaf Console with:
+
+[source]
+----
+$ ssh -p 8101 admin@localhost
+...
+admin@opennms()> feature:install flows
+admin@opennms()> config:property-set class-name org.opennms.netmgt.telemetry.listeners.udp.UdpListener
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc]: These 3 `config:` commands are not necessary.

Comment (indigo423) [reply opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/netflow5-interface.adoc]: fixed as well.

**Comment by indigo423** [comment]
branch maintenance and updated to latest features/drift
